pom.xml

```diff
@@ -274,6 +241,40 @@
             </execution>
           </executions>
         </plugin>
+        <plugin>
+          <groupId>org.jacoco</groupId>
+          <artifactId>jacoco-maven-plugin</artifactId>
+          <executions>
+            <execution>
+              <id>jacoco-prepare-agent</id>
+              <goals>
+                <goal>prepare-agent</goal>
+              </goals>
+              <configuration>
+                <append>true</append>
+                <destFile>${jacoco.exec.file}</destFile>
+                <excludes>
+                  <!-- Instrumentation of classes generated by antlr fails due to exceeding method size limit. -->
+                  <exclude>*Lexer</exclude>
+                  <!--
+                    JaCoCo adds a field to every class and as a result tests checking fields on test data classes fail.
+                  -->
+                  <exclude>org.optaplanner.core.impl.testdata.domain.*</exclude>
+                  <exclude>*TestdataAbstractSolutionBasedSolution</exclude>
+                </excludes>
+                <propertyName>jacoco.agent.argLine</propertyName>
+              </configuration>
+            </execution>
+          </executions>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-surefire-plugin</artifactId>
+          <configuration>
+            <!-- Append jacoco.agent.argLine property populated by JaCoCo's prepare-agent goal. -->
+            <argLine>@{jacoco.agent.argLine}</argLine>
+          </configuration>
+        </plugin>
```

Comment:
@rsynek Maybe we can keep this surefire plugin override in the run-code-coverage profile, so we don't need the new jacoco.agent.argLine property in <properties/>?

Reply:
Meanwhile, I ran into a reincarnation of [1], which makes it impossible to run/debug individual tests from IntelliJ Idea unless you change its maven settings to avoid including argLine to Surefire.

[1] https://youtrack.jetbrains.com/issue/IDEA-143187?_ga=2.118432521.1268541826.1581583492-754822794.1502997082